src/Tempest/Container/src/Sometimes.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Container;
+
+use Attribute;
+
+/**
+ * Add this to an attribute that has #[Inject] or a constructor parameter to indicate
+ * that your class might not always use this dependency.
+ * The container may then decide to do lazy initialization
+ */
+#[Attribute(Attribute::TARGET_PROPERTY | Attribute::TARGET_PARAMETER)]
+final readonly class Sometimes
```

Comment:
I wonder about the name. `Sometimes` makes sense, though I wonder of simply calling it `Lazy` would be more clear to people.

Reply:
I agree, the reason I didn't choose `Lazy` was that it sounds like an imperative; ie it must be lazy.
I'd prefer `ShouldBeLazy`, but in the end I don't care too much. 

So I suggest you decide on the name!

Reply:
Let's go with `Lazy` 👍